jetty-build-support/src/main/java/org/eclipse/jetty/toolchain/shade/MergeFilesTransformer.java

```diff
@@ -0,0 +1,103 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2018 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.toolchain.shade;
+
+import org.apache.maven.plugins.shade.relocation.Relocator;
+import org.apache.maven.plugins.shade.resource.ResourceTransformer;
+import org.apache.maven.shared.utils.StringUtils;
+import org.codehaus.plexus.util.IOUtil;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+
+public class MergeFilesTransformer
+    implements ResourceTransformer
+{
+
+    private List<String> files;
+
+    private String resource;
+
+    /**
+     * if the content of the resource must be preserve
+     * <code>true</code> per default
+     */
+    private boolean preserveResourceContent = true;
+
+    private ByteArrayOutputStream data = new ByteArrayOutputStream();
+
+    public MergeFilesTransformer()
+    {
+        //no op
+    }
+
+    public MergeFilesTransformer( List<String> files, String resource, boolean preserveResourceContent )
```

Comment:
A javadoc comment on the purpose of the args would be very useful in working out how to use this transformer. For example, what are the files argument - are they absolute locations on disk somewhere or are they relative to inside the artifacts that the shade plugin has been configured with?